presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryClient.java

```diff
@@ -0,0 +1,190 @@
+        this.viewMaterializationDataset = requireNonNull(config.getViewMaterializationDataset(), "viewMaterializationDataset is null");
+    }
+
+    public TableInfo getTable(TableId tableId)
+    {
+        TableId bigQueryTableId = tableIds.get(tableId);
+        Table table = bigQuery.getTable(bigQueryTableId != null ? bigQueryTableId : tableId);
+        if (table != null) {
+            tableIds.putIfAbsent(tableId, table.getTableId());
+            datasetIds.putIfAbsent(toDatasetId(tableId), toDatasetId(table.getTableId()));
+        }
+        return table;
+    }
+
+    private DatasetId toDatasetId(TableId tableId)
+    {
+        return DatasetId.of(tableId.getProject(), tableId.getDataset());
+    }
+
+    protected String getProjectId()
+    {
+        return bigQuery.getOptions().getProjectId();
+    }
+
+    protected Iterable<Dataset> listDatasets(String projectId)
+    {
+        final Iterator<Dataset> datasets = bigQuery.listDatasets(projectId).iterateAll().iterator();
+        return () -> Iterators.transform(datasets, this::addDataSetMappingIfNeeded);
+    }
+
+    protected Iterable<Table> listTables(DatasetId datasetId, TableDefinition.Type... types)
+    {
+        Set<TableDefinition.Type> allowedTypes = ImmutableSet.copyOf(types);
+        DatasetId bigQueryDatasetId = datasetIds.getOrDefault(datasetId, datasetId);
+        Iterable<Table> allTables = bigQuery.listTables(bigQueryDatasetId).iterateAll();
+        allTables.forEach(table -> addTableMappingIfNeeded(bigQueryDatasetId, table));
+        return StreamSupport.stream(allTables.spliterator(), false)
+                .filter(table -> allowedTypes.contains(table.getDefinition().getType()))
+                .collect(toImmutableList());
+    }
+
+    private void addTableMappingIfNeeded(DatasetId datasetID, Table table)
+    {
+        TableId bigQueryTableId = table.getTableId();
+        TableId prestoTableId = TableId.of(datasetID.getProject(), datasetID.getDataset(), createTableName());
+        tableIds.putIfAbsent(bigQueryTableId, prestoTableId);
+    }
+    private Dataset addDataSetMappingIfNeeded(Dataset dataset)
+    {
+        DatasetId bigQueryDatasetId = dataset.getDatasetId();
+        DatasetId prestoDatasetId = DatasetId.of(bigQueryDatasetId.getProject(), bigQueryDatasetId.getDataset().toLowerCase(ENGLISH));
+        datasetIds.putIfAbsent(prestoDatasetId, bigQueryDatasetId);
+        return dataset;
+    }
+
+    protected TableId createDestinationTable(TableId tableId)
+    {
+        String project = viewMaterializationProject.orElse(tableId.getProject());
+        String dataset = viewMaterializationDataset.orElse(tableId.getDataset());
+        DatasetId datasetId = mapIfNeeded(project, dataset);
+        return TableId.of(datasetId.getProject(), datasetId.getDataset(), createTableName());
+    }
+
+    private String createTableName()
+    {
+        String tablePrefix = "_pbc_";
```

Comment:
let's define _pbc_ as a constant string in this class, or in a constant class

Reply:
ok